src/net_processing.cpp

```diff
@@ -4383,7 +4383,7 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
             // Only actively request headers from a single peer, unless we're close to today.
             if ((nSyncStarted == 0 && fFetch) || pindexBestHeader->GetBlockTime() > GetAdjustedTime() - 24 * 60 * 60) {
                 state.fSyncStarted = true;
-                state.nHeadersSyncTimeout = count_microseconds(current_time) + HEADERS_DOWNLOAD_TIMEOUT_BASE + HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * (GetAdjustedTime() - pindexBestHeader->GetBlockTime())/(consensusParams.nPowTargetSpacing);
+                state.m_headers_sync_timeout = current_time + HEADERS_DOWNLOAD_TIMEOUT_BASE + HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * (GetAdjustedTime() - pindexBestHeader->GetBlockTime())/consensusParams.nPowTargetSpacing;
```

Comment:
Actually I think this changes the rounding calculation: `HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER` used to be 1000, in us. Now its 1 in milliseconds. The calculation `HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * x / y` used to be in us precision, but now it is only in ms precision. So there is now a bit more of a rounding error than there was before

Reply:
How so? GetAdjustedTime is divided by the time it takes for a header, so the result it the number of headers (no unit).

Multiplying a (no unit) with 1000us will give exactly (a)ms, just like multplying a (no unit) with 1ms will give exactly (a)ms.

This is correct

Reply:
Could add `(` and `)` to clarify this:

```suggestion
                state.m_headers_sync_timeout = current_time + HEADERS_DOWNLOAD_TIMEOUT_BASE + HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * ((GetAdjustedTime() - pindexBestHeader->GetBlockTime())/consensusParams.nPowTargetSpacing);
```

Reply:
The original calculation order was `(HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * ((GetAdjustedTime() - pindexBestHeader->GetBlockTime())) / consensusParams.nPowTargetSpacing`, so `(1000 * x) / y`. The PR has basically changed this into `(1 * x) / y`. I don't think this is a problem, but the result might be off by up to 999us: https://godbolt.org/z/TEYood

Reply:
not if you assume that x/y is in ℕ

Reply:
oh wait, it isn't. Missed that

Reply:
Awesome catch, @martinus ! I've changed `HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER` to be `1000us` instead of `1ms` to keep things consistent. The change may not be a problem, but I don't see a reason to risk it for moving to chrono types.

Reply:
I've converted to microseconds, but left the parenthesis the same as before because it felt more intuitive to be multiplying HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER with something that was a measure of number of headers. C++ operator precedence for `*` and `/` is left -to-right so the math shouldn't change afaict.

Reply:
Hmm, I'd still suggest adding a comment explaining the conversion to `std::chrono::microseconds` -- the usual point of chrono types is that it does the right conversions behind the scenes, so the fact it's getting done manually is weird and could use an explanation I think.

Reply:
Done.